Remove non-public OutputConfigurator usage

SethSmucker opened the pull request:
## Summary
Remove usage of non-public OutputConfigurator class.

## Files Changed
- `CBMutationOutputFormatter.java` - Replace `OutputConfigurator.getClientProperties()` with direct property building
- `ArrayIterator.java` - Remove unused SortedMapIterator import (was only in Javadoc)

## Approach
Build Accumulo client properties directly using the public API instead of relying on non-public OutputConfigurator:
```java
return Accumulo.newClientProperties()
        .to(AccumuloHelper.getInstanceName(conf), AccumuloHelper.getZooKeepers(conf))
        .as(AccumuloHelper.getUsername(conf), 
            new PasswordToken(new String(AccumuloHelper.getPassword(conf))))
        .build();
```

Fixes #3323
Part of #2443

SethSmucker committed:
Remove non-public OutputConfigurator usage

Replace OutputConfigurator.getClientProperties() with direct property
building using the public Accumulo API.

Also remove unused SortedMapIterator import from ArrayIterator.java
(was only referenced in Javadoc).

Fixes #3323
Part of #2443

foster33 commented on warehouse/query-core/src/main/java/datawave/query/iterator/logic/ArrayIterator.java:
This does not seem to belong in this PR. Could it be included in #3322 ?

foster33 commented:
There seems to be an additional use of `org.apache.accumulo.hadoopImpl.mapreduce.lib.OutputConfigurator` in `CBMutationOutputFormatterTest`

foster33 committed:
Merge branch 'integration' into task/remove-outputconfigurator

foster33 committed:
Remove other usage of OutputConfigurator

foster33 committed:
Simplify test changes

lbschanno commented on warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/CBMutationOutputFormatter.java:
```diff
@@ -28,16 +27,13 @@ public RecordWriter<Text,Mutation> getRecordWriter(TaskAttemptContext attempt) t
         //@formatter:off
-        // Convert DW's connection conf keys into Accumulo-compatible keys as required
-        clientProps.putAll(Accumulo.newClientProperties()
+        // Build client properties directly from DataWave configuration
```
The original implementation loads any client properties via OutputConfiguration.getClientProperties() before updating the properties with the customizations via `clientProps.putAll()`. As your changes stand now, you are simply returning the customizations, and not the full client properties. We need to ensure that we're loading all the client properties that we should be before returning.

SethSmucker committed:
Preserve pre-configured client properties when building Accumulo client

Address reviewer feedback: the previous implementation only returned
DataWave connection properties, losing any pre-configured client
properties (e.g., batch writer settings from shell scripts).

Now reads from AccumuloOutputFormat.ClientOpts.ClientProps configuration
key first, then overlays DataWave's connection configuration on top.
This matches the original behavior while removing the non-public
OutputConfigurator API dependency.

SethSmucker committed:
Merge integration into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator

foster33 committed:
Merge branch 'integration' into task/remove-outputconfigurator

SethSmucker committed:
Merge remote-tracking branch 'origin/integration' into task/remove-outputconfigurator
